uaclient/util.py

```diff
@@ -72,16 +68,12 @@ def encode_text(text, encoding='utf-8'):
     return text.encode(encoding)
 
 
-def is_container():
+def is_container(run_path='/run'):
     """Checks to see if this code running in a container of some sort"""
-    for helper in CONTAINER_TESTS:
-        try:
-            # try to run a helper program. if it returns true/zero
-            # then we're inside a container. otherwise, no
-            subp(helper)
-            return True
-        except (IOError, OSError):
-            pass
+    for filename in ('container_type', 'systemd/container'):
+        path = os.path.join(run_path, filename)
+        if os.path.exists(path):
+            return bool(load_file(path).strip())
```

Comment:
I _think_ with both of these files, their existence is sufficient to confirm we're in a container; do we need to read them?

Reply:
fixed and pushed is_container changes to first attempt systemd-detect-virt and fall back to presence of said files per your review.